__tests__/components/Button.test.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Button } from '@/components/Button';
+Button;
+
+describe('Button', () => {
+    it('should render', () => {
+        const handleOnClickMock = jest.fn();
+        const { container } = render(
+            <Button
+                type="button"
+                className="w-full md:w-auto bg-gray-200 px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                onClick={handleOnClickMock}
+                children="generate"
+            />
+        );
+        expect(container).toMatchSnapshot();
+        expect(container).toHaveTextContent('generate');
+    });
+
+    test('should render button with the type submit', () => {
+        const handleOnClickMock = jest.fn();
+        const { getByRole } = render(
+            <Button
+                type="submit"
+                className="w-full md:w-auto bg-gray-200 px-4 md:px-8 py-3 hover:bg-gray-300 mt-2 md:mt-0 md:rounded-none"
+                onClick={handleOnClickMock}
+                children="Log in"
+            ></Button>
```

Comment:
As you're passing the `children` prop, you don't need the closing tag for the Button component

Reply:
Done.